tests/build-and-run-tests.sh

```diff
@@ -28,14 +28,23 @@ function ciSkip {
     exit 0
 }
 
+# Finds the branching point of two commits. 
+# For example, let B and D be two commits, and their ancestry graph as A -> B, A -> C -> D. 
+# Given commits B and D, it returns A. 
+function getLowestCommonAncestorCommitHash {
```

Comment:
I changed it to `getBranchingPoint`, because `getAncestorCommit` wouldn't be correct when there are multiple commits in a branch to be merged (which is a reasonable case).

Consider when `A -> B -> C` commit ancestor graph. The branching point of C is A, but the ancestor of C is B.